# Exercise 9.3

## Part (a)

The frequentist estimator is simply the value of $\pi$ observed in the sample which is $\hat{\pi_f}=\frac{11}{116}\approx0.095$.

## Part (b)

The posterior distribution $g(\pi\mid y)$ is simply $beta(1+11,10+(116-11)) = beta(12, 115)$.

## Part (c)

$$\text{E}[\pi\mid y] = \frac{12}{12+115} \approx 0.094$$
$$\text{Var}[\pi \mid y] = \frac{12\times115}{(12+115)^2(12+115+1)}\approx0.00067$$
The Bayesian estimator is simply the expected value: $\hat{\pi_b} = E[\pi \mid y] \approx 0.094$.

## Part (d)

The 95% credible interval is
$$\text{E}[\pi\mid y] \pm 1.96\times\sqrt{\text{Var}[\pi\mid y]} \approx (0.044,0.145).$$

## Part (e)

Since $\pi = 0.10$ lies within the 95% credible interval above, we fail to reject the null hypothesis.

# DBDA Exercise 10.1

## Part (a)

In [3]:
pD = function(z,N,w,k) { exp( lbeta(z+w*(k-2)+1,N-z+(1-w)*(k-2)+1) - lbeta(w*(k-2)+1,(1-w)*(k-2)+1) ) }

In [4]:
pOdds = (pD(7, 10, 0.25,6) * 0.5)/(pD(7, 10, 0.75,6) * 0.5)
pM1 = pOdds/(1 + pOdds)
pM2 = 1 - pM1

In [5]:
cat("Posterior Model 1: ",pM1,"\n")
cat("Posterior Model 2: ",pM2,"\n")

Posterior Model 1:  0.25 
Posterior Model 2:  0.75 


A formal analysis reveals that the posterior probabilities for each factory are 0.25 and 0.75 (i.e. there is a 75% probability that the coin came from the second factory given the data).

## Part (b)

In [12]:
pOdds = (pD(7, 10, 0.25,202) * 0.5)/(pD(7, 10, 0.75,202) * 0.5)
pM1 = pOdds/(1 + pOdds)
pM2 = 1 - pM1

In [13]:
cat("Posterior Model 1: ",pM1,"\n")
cat("Posterior Model 2: ",pM2,"\n")

Posterior Model 1:  0.0159572 
Posterior Model 2:  0.9840428 


A formal analysis reveals that the posterior probabilities for each model are around 0.016 and 0.984 (i.e. there is a 98.4% probability that the coin came from the second factory given the data).

## Part (c)

Since the $\kappa$ for the second case was so high, it means that the second factory produces coins with a bias of very close 0.75 a lot of the time. Additionally, the first factory produces coins with a bias of 0.25 a lot of the time. Therefore a coin that flips 7 heads out of 10 almost surely was produced from the second factory.